client/src/main.rs

```diff
@@ -142,6 +183,299 @@ fn main() -> Result<()> {
+ * IO size to the largest possible.
+ */
+async fn baloon_workload(guest: &Arc<Guest>) -> Result<()> {
+    /*
+     * These query requests have the side effect of preventing the test from
+     * starting before the upstairs is ready.
+     */
+    let block_size = guest.query_block_size();
+    let extent_size = guest.query_extent_size();
+    let total_size = guest.query_total_size();
+    let total_blocks = (total_size / block_size) as usize;
+
+    println!(
+        "Baloon workload starting with  es: {}  bs:{}  ts:{}  tb:{}",
+        extent_size, block_size, total_size, total_blocks
+    );
+
+    /*
+     * Create an array that tracks the number of writes to each block, so
+     * we can know what to expect for reads.
+     */
+    let mut write_count = vec![0_u32; total_blocks];
+
+    for block_index in 0..total_blocks {
+        /*
+         * Loop over all the IO sizes (in blocks) that an IO can
+         * have, given our starting block and the total number of blocks
+         * We always have at least one block, and possibly more.
+         */
+        for size in 1..=(total_blocks - block_index) {
+            /*
+             * Update the write count for all blocks we plan to write to.
+             */
+            for i in 0..size {
+                write_count[block_index + i] += 1;
+            }
+
+            let vec = fill_vec(block_index, size, &write_count, block_size);
+            let data = Bytes::from(vec);
+            /*
+             * Convert block_index to its byte value.
+             */
+            let offset = (block_index as u64 * block_size) as u64;
+
+            let mut waiter = guest.write(offset, data);
+            waiter.block_wait();
+
+            let mut waiter = guest.flush();
+            waiter.block_wait();
+
+            let length: usize = size * block_size as usize;
+            let vec: Vec<u8> = vec![255; length];
+            let data = crucible::Buffer::from_vec(vec);
+            let mut waiter = guest.read(offset, data.clone());
+            waiter.block_wait();
+
+            let dl = data.as_vec().to_vec();
+            if !validate_vec(dl.clone(), block_index, &write_count, block_size)
+            {
+                bail!("Error at {}", block_index);
+            }
+        }
+    }
+
+    print_write_count(write_count, total_blocks, extent_size);
+    Ok(())
+}
+
+/*
+ * Print out the contents of the write count vector where each line
+ * is an extent and each column is a block.
+ */
+fn print_write_count(
+    write_count: Vec<u32>,
+    total_blocks: usize,
+    extent_size: u64,
+) {
+    println!(" Write IO count to each extent");
+    println!("###############################");
+    for (i, wc) in write_count.iter().enumerate().take(total_blocks) {
+        print!("{:5} ", wc);
+        if (i + 1) % (extent_size as usize) == 0 {
+            println!();
+        }
+    }
+}
+
+/*
+ * Generate a random offset and length, and write to then read from
```

Comment:
this is what hammer does :) should we fold hammer into this?

Reply:
I got the feeling hammer was a bit better test than this, but maybe in the long run they can come together.  I think hammer uses a different interface and that has value as well.  

And, we should at some point come up with a more formal test plan for Crucible, but it's probably beyond the scope of this PR.